firebase-firestore/src/main/java/com/google/firebase/firestore/model/value/ProtoValues.java

```diff
@@ -157,8 +164,8 @@ private static boolean objectEquals(Value left, Value right) {
   }
 
   /** Returns true if the Value list contains the specified element. */
-  public static boolean contains(List<Value> haystack, Value needle) {
-    for (Value haystackEl : haystack) {
+  public static boolean contains(ArrayValueOrBuilder haystack, Value needle) {
```

Comment:
Nearly all callers check the array type before calling this. It seems like it would be worth adding an alternative like this:

```java
public static boolean arrayContains(Value haystack, Value needle) {
  return isArray(haystack) && contains(haystack.getArrayValue(), needle);
}
```

Reply:
Only ArrayContainsFilter does the check right at the callsite. All other filters verify the value in the constructor, which is IMHO the better thing to do since the assert will include a stacktrace that should indicate where the incorrect value was accessed.

Furthermore, ArrayTransformOperation doesn't have a Value, it only operates on ArrayValue.Builder. For that reason, I left this as is for now.